app/scripts/controllers/network/create-network-client.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  createAsyncMiddleware,
+  createScaffoldMiddleware,
+  JsonRpcEngine,
+  mergeMiddleware,
+  JsonRpcMiddleware,
+} from 'json-rpc-engine';
+import {
+  createBlockCacheMiddleware,
+  createBlockRefMiddleware,
+  createBlockRefRewriteMiddleware,
+  createBlockTrackerInspectorMiddleware,
+  createInflightCacheMiddleware,
+  createFetchMiddleware,
+  createRetryOnEmptyMiddleware,
+} from '@metamask/eth-json-rpc-middleware';
+import {
+  providerFromEngine,
+  providerFromMiddleware,
+  SafeEventEmitterProvider,
+} from '@metamask/eth-json-rpc-provider';
+import { createInfuraMiddleware } from '@metamask/eth-json-rpc-infura';
+import type { Hex } from '@metamask/utils/dist';
+import { PollingBlockTracker } from 'eth-block-tracker/dist';
+import { SECOND } from '../../../../shared/constants/time';
+import {
+  BUILT_IN_INFURA_NETWORKS,
+  BuiltInInfuraNetwork,
+} from '../../../../shared/constants/network';
+
+export enum NetworkClientType {
+  Custom = 'custom',
+  Infura = 'infura',
+}
+
+type CustomNetworkConfiguration = {
+  chainId: Hex;
+  rpcUrl: string;
+  type: NetworkClientType.Custom;
+};
+
+type InfuraNetworkConfiguration = {
+  network: BuiltInInfuraNetwork;
+  infuraProjectId: string;
+  type: NetworkClientType.Infura;
+};
+
+/**
+ * Create a JSON RPC network client for a specific network.
+ *
+ * @param networkConfig - The network configuration.
+ * @returns
+ */
+export function createNetworkClient(
+  networkConfig: CustomNetworkConfiguration | InfuraNetworkConfiguration,
+): { provider: SafeEventEmitterProvider; blockTracker: PollingBlockTracker } {
+  /* eslint-disable @typescript-eslint/no-require-imports,@typescript-eslint/no-shadow */
+  const fetch = global.fetch || require('node-fetch');
```

Comment:
We're in a similar situation here; we can drop the fallback because `fetch` is supported by all of our supported browsers. We have it setup in unit tests already as well.

Reply:
removed here: https://github.com/MetaMask/metamask-extension/pull/17652/commits/2c67287e814d42f8c80161560ed966b4008e924f